### PR TITLE
core: Replace `on_exit_frame` with iteration over Loaders

### PR DESCRIPTION
**Pull request by Aaron1011**
This removes the need to traverse the entire display object tree - we instead just try to fire events (when ready) on corresponding `MovieClip`s for our `Loader::Movie` instances

### CODE REVIEW

**Comment by wsxarcher** [comment]
Seems related to this
https://github.com/ruffle-rs/ruffle/issues/16490

File: core/src/display_object.rs [hunk 1]
```diff
@@ -2066,19 +2067,12 @@ pub trait TDisplayObject<'gc>:
 
     /// Emit an `exitFrame` broadcast event.
     fn exit_frame(&self, context: &mut UpdateContext<'gc>) {
-        let exit_frame_evt = Avm2EventObject::bare_default_event(context, "exitFrame");
+        let exit_frame_evt: Avm2Object<'_> =
```
Comment (adrian17) [inline core/src/display_object.rs:2070]: (why was this change needed?)

File: core/src/avm2/object/loaderinfo_object.rs [hunk 2]
```diff
@@ -264,7 +264,7 @@ impl<'gc> LoaderInfoObject<'gc> {
         context: &mut UpdateContext<'gc>,
         status: u16,
         redirected: bool,
-    ) {
+    ) -> bool {
```
Comment (adrian17) [inline core/src/avm2/object/loaderinfo_object.rs:267]: can you add a comment on the return value?
Without some documentation, it took me a moment to understand what the `return true; } return false; } true` chain corresponds to

**Review by adrian17 — APPROVED** [review]

File: core/src/display_object.rs (hunk 1 above)
Comment (Aaron1011) [reply core/src/display_object.rs]: I accidentally inserted this with rust-analyer - removed
